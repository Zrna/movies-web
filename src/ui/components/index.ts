export * from './Box';
export * from './Button';
export * from './ErrorMessage';
export * from './FlexLayout';
export * from './Icon';
export * from './LoadingSpinner';
export * from './Text';
export * from './TextInput';
