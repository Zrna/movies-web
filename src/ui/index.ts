import * as styles from './styles';
import * as theme from './theme';

// export theme, enums, styles
export { styles, theme };
