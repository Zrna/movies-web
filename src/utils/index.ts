export * from './async';
export * from './error';
export * as validator from './validation';
