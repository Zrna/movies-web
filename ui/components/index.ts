export * from './Box';
export * from './Button';
export * from './Checkbox';
export * from './Divider';
export * from './ErrorMessage';
export * from './FlexLayout';
export * from './Icon';
export * from './Image';
export * from './LoadingSpinner';
export * from './Modal';
export * from './Only';
export * from './Search';
export * from './Text';
export * from './Textarea';
export * from './TextInput';
export * from './Toast';
