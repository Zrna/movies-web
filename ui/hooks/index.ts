export * from './dom';
export * from './modal';
