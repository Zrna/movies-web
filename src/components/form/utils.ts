import { FieldMetaState } from 'react-final-form';

import { TextInputProps } from '~/ui';

export interface FormFieldProps extends Omit<TextInputProps, 'value' | 'onChange'> {
  name: string;
  type?: string;
  label?: string;
  validate?: any;
  value?: null;
  onChange?(): null;
}

export function getFieldError(meta: FieldMetaState<any>) {
  if (meta.touched && meta.error) {
    return meta.error;
  }

  return meta.submitError;
}
