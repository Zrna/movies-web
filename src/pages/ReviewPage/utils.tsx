import { theme } from '~/ui';

export const reactionsMap: { [key: number]: { icon: theme.Icon; text: string } } = {
  0: { icon: 'reaction0', text: 'No comment' },
  1: { icon: 'reaction1', text: "I wish I hadn't watched" },
  2: { icon: 'reaction2', text: 'Boring' },
  3: { icon: 'reaction3', text: 'Not great, not terrible' },
  4: { icon: 'reaction4', text: "It's great" },
  5: { icon: 'reaction5', text: 'Loving it!' },
};
