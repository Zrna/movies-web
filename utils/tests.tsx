import { cleanup, fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'theme-ui';

import { theme } from '~/ui';

export { cleanup, fireEvent, render, userEvent };

export function renderWithTheme(Component: JSX.Element) {
  return render(<ThemeProvider theme={theme}>{Component}</ThemeProvider>);
}

export function getThemeColorString(color: theme.Color) {
  return `var(--theme-ui-colors-${color})`;
}

export function expectTextVariant(element: HTMLElement, variant: theme.TextVariant) {
  const { fontFamily, fontSize, fontStyle, fontWeight, lineHeight } = theme.text[variant];

  expect(element).toHaveStyleRule('font-family', fontFamily);
  expect(element).toHaveStyleRule('font-style', fontStyle);
  expect(element).toHaveStyleRule('font-size', fontSize);
  expect(element).toHaveStyleRule('font-weight', fontWeight.toString());
  expect(element).toHaveStyleRule('line-height', lineHeight);

  if (variant === 'label-uppercase') {
    expect(element).toHaveStyleRule('text-transform', theme.text[variant].textTransform);
  }
}

export function mockWindowMatchMedia() {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
}
