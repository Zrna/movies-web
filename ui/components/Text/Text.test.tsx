import { screen } from '@testing-library/react';

import { theme } from '~/ui';
import { expectTextVariant, renderWithTheme } from '~/utils/tests';

import { Text } from './';

describe('Text', () => {
  it('should render', () => {
    renderWithTheme(
      <Text as="p" color="primary" data-testid="text" variant="paragraph-big">
        Text Paragraph Big
      </Text>,
    );

    expect(screen.getByTestId('text')).toMatchSnapshot();
  });

  describe('variant', () => {
    it('should apply default vairant', () => {
      const defaultVariant = 'paragraph-default';
      renderWithTheme(<Text data-testid="text">Text</Text>);

      expectTextVariant(screen.getByTestId('text'), defaultVariant);
    });

    it('should apply supported variants', () => {
      const variants = Object.keys(theme.text) as theme.TextVariant[];
      variants.forEach((variant) => {
        const testId = `${variant}-testId`;
        renderWithTheme(
          <Text data-testid={testId} variant={variant}>
            Text
          </Text>,
        );

        expectTextVariant(screen.getByTestId(testId), variant);
      });
    });
  });

  describe('other', () => {
    it('should support `as` prop', () => {
      renderWithTheme(
        <Text as="h1" data-testid="text1">
          Text1
        </Text>,
      );
      renderWithTheme(
        <Text as="span" data-testid="text2">
          Text2
        </Text>,
      );
      renderWithTheme(
        <Text as="p" data-testid="text3">
          Text3
        </Text>,
      );
      renderWithTheme(
        <Text as="input" data-testid="text4" placeholder="Test placeholder" value="test" onChange={() => null} />,
      );

      const href = 'https://google.com';
      renderWithTheme(
        <Text as="a" data-testid="text5" {...{ href }}>
          Text5
        </Text>,
      );

      expect(screen.getByTestId('text1').tagName.toLowerCase()).toBe('h1');
      expect(screen.getByTestId('text2').tagName.toLowerCase()).toBe('span');
      expect(screen.getByTestId('text3').tagName.toLowerCase()).toBe('p');
      expect(screen.getByTestId('text4').tagName.toLowerCase()).toBe('input');
      expect(screen.getByTestId('text5').tagName.toLowerCase()).toBe('a');
      expect(screen.getByTestId('text5')).toHaveAttribute('href', href);
    });

    it('should support `sx` prop', () => {
      const customStyles = { textDecoration: 'underline', textOverflow: 'ellipsis', overflow: 'hidden' };
      renderWithTheme(
        <Text data-testid="text" sx={customStyles}>
          Text
        </Text>,
      );

      expect(screen.getByTestId('text')).toHaveStyle(customStyles);
    });
  });
});
