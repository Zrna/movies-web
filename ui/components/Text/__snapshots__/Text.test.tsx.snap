// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render 1`] = `
<p
  class="css-1b1m0xc-Box"
  data-testid="text"
>
  Text Paragraph Big
</p>
`;
